File: py/pe/et.py
class A(object):

    def __init__(self, a, b):
        self.a = a
        self.b = b

    def update(self, a):
        self.a = a

    def update(self, a, b):
        self.a = a
        self.b = b


a = A(1, 2)
a.update(3)
a.update(3, 4)

File: py/pe/et.php
<?php

class A
{
    private $a = null;
    private $b = null;

    function __construct($a)
    {
    	// code...
    }

    function update($a)
    {
        $this->a = $a;
    }

    function update($a, $b)
    {
        $this->a = $a;
        $this->b = $b;
    }

    function print()
    {
        printf("a = %s, b = %s", $this->a, $this->b);
    }
}

$x = new A('1');
$x->update('2');
$x->update('2', '3');
$x->print();
